exporter/skywalkingexporter/metricrecord_to_metricdata.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package skywalkingexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/skywalkingexporter"
+
+import (
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	conventions "go.opentelemetry.io/collector/model/semconv/v1.5.0"
+	metricpb "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
+)
+
+const (
+	defaultServiceInstance = "otel-collector-instance"
+)
+
+func resourceToMetricLabels(resource pdata.Resource) []*metricpb.Label {
+	labels := make([]*metricpb.Label, 0)
+	attrs := resource.Attributes()
+	attrs.Range(func(k string, v pdata.AttributeValue) bool {
+		labels = append(labels,
+			&metricpb.Label{
+				Name:  k,
+				Value: v.AsString(),
+			})
+		return true
+	})
+	return labels
+}
+
+func resourceToServiceInfo(resource pdata.Resource) (service string, serviceInstance string) {
+	attrs := resource.Attributes()
+	if serviceName, ok := attrs.Get(conventions.AttributeServiceName); ok {
+		service = serviceName.AsString()
+	} else {
+		service = defaultServiceName
+	}
+	if serviceInstanceID, ok := attrs.Get(conventions.AttributeServiceInstanceID); ok {
+		serviceInstance = serviceInstanceID.AsString()
+	} else {
+		serviceInstance = defaultServiceInstance
+	}
+	return service, serviceInstance
+}
+
+func numberMetricsToData(name string, data pdata.NumberDataPointSlice, defaultLabels []*metricpb.Label) (metrics []*metricpb.MeterData) {
+	metrics = make([]*metricpb.MeterData, 0)
```

Comment:
Will the len of the resulting `metrics` always be the same as `data.Len()`? If so, you might want to initialize this with the final size, potentially saving some resize operations.